Bottom: 12b0ace3981d06c1a29d65dfc8877bb062282b4f
Top:    e303f20481c9085e70236c012bbfa6886fd92bf1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 11:49:27 +0100

Expose draw.c internal drawing APIs


---

diff --git a/src/draw.c b/src/draw.c
index e3c79e3..1a7171e 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -92,10 +92,7 @@ static int DrawLayerGroup (int, const BoxType *);
 static void AddPart (void *);
 static void SetPVColor (PinTypePtr, int);
 static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
-static void DrawMask (int side, BoxType *);
-static void DrawPaste (int side, BoxType *);
 static void DrawRats (BoxType *);
-static void DrawSilk (int side, const BoxType *);
 
 /*--------------------------------------------------------------------------------------
  * setup color for pin or via
@@ -911,7 +908,7 @@ clearPad_callback (const BoxType * b, void *cl)
  * Draws silk layer.
  */
 
-static void
+void
 DrawSilk (int side, const BoxType * drawn_area)
 {
 #if 0
@@ -970,7 +967,7 @@ DrawMaskBoardArea (int mask_type, BoxType *screen)
 /* ---------------------------------------------------------------------------
  * draws solder mask layer - this will cover nearly everything
  */
-static void
+void
 DrawMask (int side, BoxType * screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
@@ -999,7 +996,7 @@ DrawMask (int side, BoxType * screen)
 /* ---------------------------------------------------------------------------
  * draws solder paste layer for a given side of the board
  */
-static void
+void
 DrawPaste (int side, BoxType *drawn_area)
 {
   gui->set_color (Output.fgGC, PCB->ElementColor);
diff --git a/src/draw.h b/src/draw.h
index adfee36..46f67d4 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -72,6 +72,10 @@ void EraseElementName (ElementTypePtr);
 void EraseObject (int, void *, void *);
 void LoadBackgroundImage (char *);
 
+void DrawMask (int side, BoxType *);
+void DrawSilk (int side, const BoxType *);
+void DrawPaste (int side, BoxType *);
+
 /*GdkDrawable *draw_get_current_drawable(void);*/
 
 #endif
